doc/examples/example_ecies.md

```diff
@@ -0,0 +1,378 @@
+            .setIV(std::vector<uint8_t>(IV))
+            .setPadding(mococrw::SymmetricCipherPadding::PKCS)
+            .buildEncryptor();
+};
+encBuilder.setSymmetricCipherFactoryFunction(cipherEncFunc);
+
+// 4. Set new key size
+encBuilder.setSymmetricCipherKeySize(mococrw::getSymmetricCipherKeySize(mococrw::SymmetricCipherKeySize::S_128));
+
+std::unique_ptr<mococrw::ECIESEncryptionCtx> encCtx = encBuilder.buildEncryptionCtx(pubKey);
+encCtx->update(testString);
+
+// 5. Finish encryption and extract parameters for decryption
+std::vector<uint8_t> ciphertext = encCtx->finish();
+std::vector<uint8_t> mac = encCtx->getMAC();
+std::vector<uint8_t> point = encCtx->getEphemeralKey().toECPoint(...); // See section "Ephemeral Key Encodig Format" for details
+```
+
+### Decryption
+
+```cpp
+// Ciphertext, mac, point and IV need to be available
+std::vector<uint8_t> ciphertext = { ... };
+std::vector<uint8_t> mac = { ... };
+std::vector<uint8_t> point = { ... };
+std::vector<uint8_t> IV = { ... };
+
+// 1. Read private key from PEM
+mococrw::AsymmetricKeypair privKey = mococrw::AsymmetricKeypair::readPrivateKeyFromPEM(secp384KeyPEM, "");
+
+// 2. Get type of curve from private key
+std::shared_ptr<mococrw::ECCSpec> spec;
+{
+    // Convert returned unique_ptr into shared_ptr
+    std::shared_ptr<mococrw::AsymmetricKey::Spec> spec_tmp = privKey.getKeySpec();
+
+    // Cast shared_ptr<mococrw::AsymmetricKey::Spec> into std::shared_ptr<mococrw::ECCSpec>
+    spec = std::dynamic_pointer_cast<mococrw::ECCSpec>(spec_tmp);
+}
+
+if (!spec) {
+    // Error handling
+    ...
+}
+
+// 3. Re-create ephemeral key
+mococrw::AsymmetricPublicKey ephKey = mococrw::AsymmetricPublicKey::fromECPoint(spec, point);
+
+mococrw::ECIESCtxBuilder decBuilder;
+
+// 4. Set new Symmetric Cipher Factory Function
+auto cipherDecFunc = [&IV](const std::vector<uint8_t> &key) -> std::unique_ptr<mococrw::AESCipher> {
+    return mococrw::AESCipherBuilder(mococrw::SymmetricCipherMode::CBC, mococrw::SymmetricCipherKeySize::S_128, key)
+            .setIV(IV)
+            .setPadding(mococrw::SymmetricCipherPadding::PKCS)
+            .buildDecryptor();
+};
+decBuilder.setSymmetricCipherFactoryFunction(cipherDecFunc);
+
+// 5. Set new key size
+decBuilder.setSymmetricCipherKeySize(mococrw::getSymmetricCipherKeySize(mococrw::SymmetricCipherKeySize::S_128));
+
+// 6. Set parameters for decryption
+std::unique_ptr<mococrw::ECIESDecryptionCtx> decCtx = decBuilder.buildDecryptionCtx(privKey, ephKey);
+decCtx->update(ciphertext);
+decCtx->setMAC(mac);
+
+// 7. Decrypt
+try {
+    std::vector<uint8_t> result = decCtx->finish();
+}
+catch (const MoCOCrWException &e)  {
+    std::cerr << "MAC verification failed" << std::endl;
+    ...
+}
+```
+
+## Key Derivation Function (KDF)
+
+MoCOCrW offers different key derivation functions. The following example show how to change the KDF used by ECIES. In addition, the example shows
```

Comment:
```suggestion
MoCOCrW offers different key derivation functions. The following example shows how to change the KDF used by ECIES. In addition, the example shows
```

This introduces a repetition of "examples shows how to", so maybe we can change one of them to "the example demonstrates setting …"